java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -154,7 +156,9 @@ public RewriterResult rewrite(ConstrainedTerm constrainedTerm, int bound) {
 
     }
 
-    public List<ConstrainedTerm> fastComputeRewriteStep(ConstrainedTerm subject, boolean computeOne, boolean narrowing, boolean proofFlag, int step) {
+    public List<ConstrainedTerm> fastComputeRewriteStep(ConstrainedTerm subject, boolean computeOne, boolean narrowing,
+                                                        boolean proofFlag, int step,
+                                                        ConstrainedTerm initTerm) {
```

Comment:
Not sure if it is a good idea to add new argument (only for logging purpose) to the fastComputeRewriteStep function. Can we call the logging function after ``List<ConstrainedTerm> results = computeRewriteStep(term, step, false)``, can traverse the list to print the constraint diff?

Reply:
This would degrade the usability of logs. First, `fastComputeRewriteStep` is called from 3 places. Most important is here: https://github.com/kframework/k/blob/c98afd1439d4b50ad6133ebcc51cdf1546a43104/java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java#L722-L723

Second, logs are most useful when they are printed just when the event that triggered them happened. In the current implementation, I can look at the logs generated before `New top-level constraint` to see what triggered new constraint. If `New constraint` logs would be printed later, it wouls be harder to make connection with rules and z3 queries associated with them. I'll add a comment to the argument to make it clear it's only used for logging.

Reply:
ok